tests/test_decl.py

```diff
@@ -206,6 +206,87 @@ def test_decl_can_create_simple_objects_by_passing_plain_strings(
 
         assert "NewGame++" in attrdef.values
 
+    @staticmethod
+    def test_extend_operations_change_model_objects_in_place(
+        model: capellambse.MelodyModel,
+    ) -> None:
+        root_function = model.by_uuid(ROOT_FUNCTION)
+        fnc = model.by_uuid("8833d2dc-b862-4a50-b26c-6f7e0f17faef")
+        old_parent = fnc.parent
+
+        assert fnc not in root_function.functions, "Precondition failed"
+
+        yml = f"""\
+            - parent: !uuid {ROOT_FUNCTION}
+              extend:
+                functions:
+                  - !uuid {fnc.uuid}
+            """
+
+        decl.apply(model, io.StringIO(yml))
+
+        assert fnc in root_function.functions
+        assert fnc.parent == root_function != old_parent
```

Comment:
Instead of this roundabout check at the end (where I'm not even sure if the ordering of operands is correct..? The first two might be swapped), it would IMO be cleaner to add another precondition to explicitly `assert old_parent != root_function`. (Even though technically that and the `root_function != old_parent` part here are redundant anyways, but in this case I agree with "explicit is better than implicit".)